src/main/kotlin/exercise4/task1/BankAccount.kt

```diff
@@ -32,21 +32,51 @@ package exercise4.task1
  *      The initial balance for this constructor should always be set to 0.
  */
 
+open class BankAccount(private val accountNumber: String,
+                       private val accountHolderName: String,
+                       private var balance: Double = 0.0) {
+
+    open fun deposit(amount: Double): Boolean {
+        if (amount <= 0) {
+            println("Invalid deposit amount. The amount must be non-negative.\n")
+            return false
+        }
+        balance += amount
+        return true
+    }
+
+    open fun withdraw(amount: Double): Boolean {
+        if(balance < amount) {
+            println("There are not enough funds in the account to make the requested withdrawal.\n")
+        }
```

Comment:
This will allow the user to withdraw the money even though withdrawal would cause the account to have a negative balance.

Function should return `false` if `amount` is greater than `balance`, and `balance` should stay unchanged.

Reply:
It was an oversight on my part.